internal/controller/postgrescluster/pgbackrest.go

```diff
@@ -1066,51 +1052,15 @@ func (r *Reconciler) reconcileRestoreJob(ctx context.Context,
 		volumeMounts = append(volumeMounts, walVolumeMount)
 	}
 
-	restoreJob := &batchv1.Job{
-		ObjectMeta: meta,
-		Spec: batchv1.JobSpec{
-			Template: v1.PodTemplateSpec{
-				ObjectMeta: metav1.ObjectMeta{
-					Annotations: annotations,
-					Labels:      labels,
-				},
-				Spec: v1.PodSpec{
-					Containers: []v1.Container{{
-						Command:         cmd,
-						Image:           cluster.Spec.Image,
-						Name:            naming.PGBackRestRestoreContainerName,
-						VolumeMounts:    volumeMounts,
-						Env:             []v1.EnvVar{{Name: "PGHOST", Value: "/tmp"}},
-						SecurityContext: initialize.RestrictedSecurityContext(),
-						Resources:       dataSource.Resources,
-					}},
-					RestartPolicy: v1.RestartPolicyNever,
-					Volumes:       volumes,
-				},
-			},
-		},
-	}
-
-	// Set the image pull secrets, if any exist.
-	// This is set here rather than using the service account due to the lack
-	// of propagation to existing pods when the CRD is updated:
-	// https://github.com/kubernetes/kubernetes/issues/88456
-	restoreJob.Spec.Template.Spec.ImagePullSecrets = cluster.Spec.ImagePullSecrets
+	restoreJob := &batchv1.Job{}
+	generateRestoreJobIntent(ctx, cluster, configHash, instanceName, cmd,
+		volumeMounts, volumes, dataSource, restoreJob)
 
 	restoreJob.SetGroupVersionKind(batchv1.SchemeGroupVersion.WithKind("Job"))
 	if err := errors.WithStack(r.setControllerReference(cluster, restoreJob)); err != nil {
 		return errors.WithStack(err)
 	}
```

Comment:
:question: Should these be considered part of the intent?

Reply:
Yeah I looks like it should be. I was trying to keep api calls out of this function but it looks like the set controller reference isn't a call. I can work on adding that in, should be an easy chang.